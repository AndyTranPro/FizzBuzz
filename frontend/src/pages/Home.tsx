import React from 'react';
import { Container, Typography, Box, Grid } from '@mui/material';
import GameList from '../components/GameList';
import GameForm from '../components/GameForm';
import VideogameAssetOutlinedIcon from '@mui/icons-material/VideogameAssetOutlined';

// The Home page, which displays the game form and the list of games

const Home: React.FC = () => {
    return (
        <Container>
            <Box my={4}>
                <Box
                    sx={{
                        display: 'flex',
                        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
                        padding: '15px',
                        borderRadius: '8px',
                        textAlign: 'center',
                        color: 'white',
                        justifyContent: 'space-around',
                        alignItems: 'center',
                    }}
                >
                    <VideogameAssetOutlinedIcon sx={{ fontSize: 80, marginBottom: 1 }} />
                    <Typography
                        variant="h3"
                        component="h1"
                        gutterBottom
                        sx={{
                            justifyContent: 'center',
                            WebkitBackgroundClip: 'text',
                            fontFamily: 'cursive',
                            fontWeight: 'bold'
                        }}
                    >
                        FizzBuzz Game
                    </Typography>
                    <VideogameAssetOutlinedIcon sx={{ fontSize: 80, marginBottom: 1 }} />
                </Box>
                <Grid container spacing={4} mt={4}>
                    <Grid item xs={12} md={6}>
                        <GameForm />
                    </Grid>
                    <Grid item xs={12} md={6}>
                        <GameList />
                    </Grid>
                </Grid>
            </Box>
        </Container>
    );
};

export default Home;