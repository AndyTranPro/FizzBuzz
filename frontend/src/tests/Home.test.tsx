import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../pages/Home';

jest.mock('../components/GameForm', () => () => <div>Mocked GameForm</div>);
jest.mock('../components/GameList', () => () => <div>Mocked GameList</div>);

describe('Home Component', () => {
    test('renders correctly and matches snapshot', () => {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the title "FizzBuzz Game"', () => {
        render(<Home />);
        expect(screen.getByText('FizzBuzz Game')).toBeInTheDocument();
    });

    test('renders GameForm and GameList components', () => {
        render(<Home />);
        expect(screen.getByText('Mocked GameForm')).toBeInTheDocument();
        expect(screen.getByText('Mocked GameList')).toBeInTheDocument();
    });
});