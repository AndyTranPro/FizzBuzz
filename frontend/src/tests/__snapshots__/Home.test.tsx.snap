// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-zitybv"
    >
      <div
        class="MuiBox-root css-wua46y"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bmzd0o-MuiSvgIcon-root"
          data-testid="VideogameAssetOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 6H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2m0 10H3V8h18zM6 15h2v-2h2v-2H8V9H6v2H4v2h2z"
          />
          <circle
            cx="14.5"
            cy="13.5"
            r="1.5"
          />
          <circle
            cx="18.5"
            cy="10.5"
            r="1.5"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-umzl7p-MuiTypography-root"
        >
          FizzBuzz Game
        </h1>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bmzd0o-MuiSvgIcon-root"
          data-testid="VideogameAssetOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 6H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2m0 10H3V8h18zM6 15h2v-2h2v-2H8V9H6v2H4v2h2z"
          />
          <circle
            cx="14.5"
            cy="13.5"
            r="1.5"
          />
          <circle
            cx="18.5"
            cy="10.5"
            r="1.5"
          />
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-19ezdg1-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div>
            Mocked GameForm
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div>
            Mocked GameList
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
